activate

```diff
@@ -0,0 +1,86 @@
+#!/bin/bash
+# This script sends a one-off activation message (on first boot-up) to
+# Endless, in order to discover how many Endless OS systems are started
+# across the world.
+# 
+# Note that this does not send any identifiable user data: it only transmits the
+# originally installed operating system version, the current version, the
+# machine vendor & product name. When the first activation message is sent, it
+# contains the machine serial number, but this is not sent again so cannot be
+# used for tracking purposes.
+# 
+# (C) 2014 Canonical Ltd.
+# (C) 2016 Endless Mobile Inc.
+# License: GPL v2 or later
+
+# https://home.endlessm.com/v1/activate
+# 
+# Arguments in the query string will be:
+# 
+#  image (the image name, from the xattr on /sysroot)
+#  release (from /etc/os-release)
+#  vendor (from DMI / device tree)
+#  product (from DMI / device tree)
+#  serial (from DMI / Endless mfgdata)
+#  live (true if this is a live USB, false if not)
+
+set -e
+
+DIR=/var/lib/eos-phone-home
+
+if [ ! -d $DIR -o ! -w $DIR ]; then
+    exit 1
+fi
+
+ACTIVATEDFILE=$DIR/activated
+
+# Give me a ping, Vasili. One ping only, please.
+if [ -e $ACTIVATEDFILE ]; then
+    exit 0
+fi
+
+SCRIPT=https://home.endlessm.com/v1/activate
+
+image=$(attr -q -g eos-image-version /sysroot) || image=''
+
+eval $(grep ^VERSION= /etc/os-release)
+release=${VERSION}
+
+if [ -d /sys/class/dmi/id ]; then
+    # do we want to do this? hmm...
+    #if [ "$vendor" == "LENOVO" ]; then
+    #  product_path="/sys/class/dmi/id/product_version"
+    #else
+    product_path="/sys/class/dmi/id/product_name"
+    #fi
+    
+    vendor=$(< /sys/class/dmi/id/sys_vendor) || vendor=''
+    product=$(< "$product_path" ) || product=''
+    serial=$(< /sys/class/dmi/id/product_serial) || serial=''
+else
+    compat=$(cut -d '' -f1 < /proc/device-tree/compatible) || compat=''
+    vendor=$(echo ${compat} | cut -d, -f1)
+    product=$(echo ${compat} | cut -d, -f2)
+    serial=$(< /sys/class/endless_mfgdata/entries/SSN) || serial=''
+fi
+
+vendor="${vendor%"${vendor##*[![:space:]]}"}"
+product="${product%"${product##*[![:space:]]}"}"
+serial="${serial%"${serial##*[![:space:]]}"}"
+
+# FIXME implement actual detection here
+live=false
+
+# report in
+if ! out=$(wget -O - -q "$SCRIPT?image=$image&release=$release&vendor=$vendor&product=$product&serial=$serial&live=$live"); then
```

Comment:
I made the server API handle these as PUT params with JSON rather than a string since that's the proper RESTful way to create a record on the server over http APIs and we may have some character encoding issues if we do things so we might need to change this.